openwisp_controller/notifications/handlers.py

```diff
@@ -0,0 +1,25 @@
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from openwisp_notifications.signals import notify
+from swapper import load_model
+
+from openwisp_controller.config.signals import config_status_changed
+
+Config = load_model('config', 'Config')
+Device = load_model('config', 'Device')
+
+
+@receiver(
+    config_status_changed,
+    sender=Config,
+    dispatch_uid='config_status_error_notification',
+)
+def config_status_error_notification(sender, instance, **kwargs):
+    if instance.status == 'error':
+        notify.send(sender=instance, type='config_error', target=instance.device)
+
+
+@receiver(post_save, sender=Device, dispatch_uid='device_registered_notification')
+def device_registered_notification(sender, instance, created, **kwargs):
+    if created:
```

Comment:
There's a problem with this: we will generate notifications also when a device is added from the admin, which is weird.. because the registration is a different concept.

Can you help me to add a signal for the registration?
The place to emit the signal is here: https://github.com/openwisp/openwisp-controller/blob/master/openwisp_controller/config/controller/views.py#L344
The parameter can be simply the device instance, the docs section to update is this one: https://github.com/openwisp/openwisp-controller#signals

Once the new signal is added, you can connect this receiver function to it.